libs/sdk-core/src/persist/settings.rs

```diff
@@ -50,13 +50,11 @@ impl SqliteStorage {
     }
 
     pub fn set_lsp_id(&self, lsp_id: String) -> Result<()> {
-        self.update_setting("lsp".to_string(), lsp_id)?;
-        Ok(())
+        self.update_setting("lsp".to_string(), lsp_id)
```

Comment:
if we change `update_settings` to return PersistenceFailure then it would propagate all the way up.

Reply:
Done